src/index.ts

```diff
@@ -0,0 +1,45 @@
+import { v4 as uuidv4 } from 'uuid';
+
+type AlgorithmFunction = () => string;
+
+const algorithms: Record<string, AlgorithmFunction> = {
+  uuid4: uuidv4,
+};
+
+function replacingAlgorithm(
+  memoizationTable: Record<string, string>,
+  _: unknown,
+  uriBase: string,
+  uuidAlgorithm: string,
+  originalUuid: string,
+) {
+  if (memoizationTable[`${uriBase}${originalUuid}`]) {
+    return memoizationTable[`${uriBase}${originalUuid}`];
+  } else {
+    const algorithm = algorithms[uuidAlgorithm] || uuidv4;
```

Comment:
Should it default to `uuidv4`? It seems like this could hide problems from missing algorithms.